### ISSUE TITLE
comments

### ISSUE THREAD

**Issue by apgarcia749** [flaurencyac/CitySlicker#5]
two overall comments:
1) excellent job breaking things down into short simple functions! 
2) the scope of your functions/variables should usually be as small as possible. so if it's possible to make a function protected/private then do so
for ex: https://github.com/flaurencyac/CitySlicker/blob/50e9a55591e6be8603ea122421b0bc9b74ab95d8/app/src/main/java/com/codepath/cityslicker/ui/profile/ProfileFragment.java#L171
seems like this could be private (going off of github's list of usages)

https://github.com/flaurencyac/CitySlicker/blob/50e9a55591e6be8603ea122421b0bc9b74ab95d8/app/src/main/java/com/codepath/cityslicker/Utilities.java#L50
great idea to add the tag to make debugging easier!

https://github.com/flaurencyac/CitySlicker/blob/50e9a55591e6be8603ea122421b0bc9b74ab95d8/app/src/main/java/com/codepath/cityslicker/ui/profile/ProfileFragment.java#L192
does this handle the possible null bitmap?

https://github.com/flaurencyac/CitySlicker/blob/50e9a55591e6be8603ea122421b0bc9b74ab95d8/app/src/main/java/com/codepath/cityslicker/ui/explore/ExploreFragment.java#L129
does this handle if `getDeviceLocation` leads to an exception? and does L133 still work in that case?

https://github.com/flaurencyac/CitySlicker/blob/50e9a55591e6be8603ea122421b0bc9b74ab95d8/app/src/main/java/com/codepath/cityslicker/ui/compose/ComposeFragment.java#L354
why? and so if the list has 0 collaborators then it will have size 1 and won't need the final element dropped?

https://github.com/flaurencyac/CitySlicker/blob/50e9a55591e6be8603ea122421b0bc9b74ab95d8/app/src/main/java/com/codepath/cityslicker/fragments/AddToTripDialogFragment.java#L132
nit: `PARCEL_NAME`

https://github.com/flaurencyac/CitySlicker/blob/50e9a55591e6be8603ea122421b0bc9b74ab95d8/app/src/main/java/com/codepath/cityslicker/fragments/AddToTripDialogFragment.java#L133
nit: just declare a context for the entire fragment

https://github.com/flaurencyac/CitySlicker/blob/50e9a55591e6be8603ea122421b0bc9b74ab95d8/app/src/main/java/com/codepath/cityslicker/fragments/POIDialogFragment.java#L114
nit: `PLACE`

https://github.com/flaurencyac/CitySlicker/blob/50e9a55591e6be8603ea122421b0bc9b74ab95d8/app/src/main/java/com/codepath/cityslicker/fragments/POIDialogFragment.java#L115
nit: just declare a context for the entire fragment

https://github.com/flaurencyac/CitySlicker/blob/50e9a55591e6be8603ea122421b0bc9b74ab95d8/app/src/main/java/com/codepath/cityslicker/adapters/TripsAdapter.java#L146
are you sure this is what you want? for every spot you're starting another activity?
one thing that _might_ neaten this up is if you use `Futures` https://www.baeldung.com/java-future
basically they are wrappers around objects and you can execute on them once they exist. usually these are used on things that might need to be done async. another alternative is to add a `synchronous {}` block here
